### PR TITLE
Bug Fixes with UserDiscoverer Lag

### PR DESCRIPTION
**Pull request by hsyyid**
Previously, it would try to get a `GameProfile`, but it would always return `null`. This would cause pretty much infinite searching of the servers dat files - causing large amounts of lag.

Through this, it gets the `GameProfile` correctly and all lag is gone. :smile:

### CODE REVIEW

File: src/main/java/org/spongepowered/common/service/user/UserDiscoverer.java [hunk 1]
```diff
@@ -45,20 +45,19 @@
...
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
 import java.util.concurrent.TimeUnit;
 
 class UserDiscoverer {
 
-    private static final Cache<UUID, User> userCache = CacheBuilder.newBuilder()
-            .expireAfterAccess(1, TimeUnit.DAYS)
-            .build();
+    private static final Cache<UUID, User> userCache = CacheBuilder.newBuilder().expireAfterAccess(1, TimeUnit.DAYS).build();
```
Comment (kashike) [inline src/main/java/org/spongepowered/common/service/user/UserDiscoverer.java]: Why this change?

File: src/main/java/org/spongepowered/common/service/user/UserDiscoverer.java [hunk 2]
```diff
@@ -182,20 +184,29 @@ private static User getFromStoredData(UUID uniqueId) {
...
         } else {
             return null;
         }
     }
 
+    private static Optional<org.spongepowered.api.profile.GameProfile> getProfileFromServer(UUID uuid) {
+        CompletableFuture<org.spongepowered.api.profile.GameProfile> gameProfile = Sponge.getServer().getGameProfileManager().get(uuid);
+
+        try {
+            if (gameProfile.get() != null) {
```
Comment (kashike) [inline src/main/java/org/spongepowered/common/service/user/UserDiscoverer.java]: Store value of `gameProfile.get()` instead of calling it twice

File: src/main/java/org/spongepowered/common/service/user/UserDiscoverer.java [hunk 3]
```diff
@@ -182,20 +184,29 @@ private static User getFromStoredData(UUID uniqueId) {
         if (dataFile == null) {
             return null;
         }
-        GameProfile profile = MinecraftServer.getServer().getPlayerProfileCache().getProfileByUUID(uniqueId);
-        if (profile != null) {
-            User user = create(profile);
-            try {
-                ((SpongeUser) user).readFromNbt(CompressedStreamTools.readCompressed(new FileInputStream(dataFile)));
```
Comment (Aaron1011) [inline src/main/java/org/spongepowered/common/service/user/UserDiscoverer.java]: What happened to this line?

File: src/main/java/org/spongepowered/common/service/user/UserDiscoverer.java [hunk 4]
```diff
@@ -77,6 +81,7 @@ static User create(GameProfile profile) {
      */
     static User findByUuid(UUID uniqueId) {
         User user = userCache.getIfPresent(uniqueId);
+
```
Comment (kashike) [inline src/main/java/org/spongepowered/common/service/user/UserDiscoverer.java]: No need for the new whitespace

File: src/main/java/org/spongepowered/common/service/user/UserDiscoverer.java [hunk 5]
```diff
@@ -196,6 +203,21 @@ private static User getFromStoredData(UUID uniqueId) {
...
+        CompletableFuture<org.spongepowered.api.profile.GameProfile> gameProfile = Sponge.getServer().getGameProfileManager().get(uuid);
+
+        try {
+            org.spongepowered.api.profile.GameProfile profile = gameProfile.get();
+            if (profile != null) {
+                return Optional.of(profile);
+            } else {
+                return Optional.empty();
+            }
+        } catch (InterruptedException | ExecutionException e) {
```
Comment (gabizou) [inline src/main/java/org/spongepowered/common/service/user/UserDiscoverer.java]: Why are you just swallowing the exception? This needs to be logged for any reason.

Comment (kenzierocks) [inline src/main/java/org/spongepowered/common/service/user/UserDiscoverer.java]: Isn't that happening below?
